tags.go

```diff
@@ -0,0 +1,42 @@
+package distribution
+
+// Tag is a named reference to a repository object. Target generally refers to
+// a manifest.
+type Tag interface {
+	Blob // io access to serialized content.
+
+	// Name returns the fully qualified name of the object. For example, if
+	// the object is in repository "stevvooe/test" and has tag "latest", the
+	// name will be "stevvooe/test:latest".
+	Name() string
+
+	// Description returns a human-readable description of the tag.
+	Description() string
+
+	// Target identifies the blob targeted by this tag.
+	Target() Descriptor
+}
+
+// TagService provides access to information about tagged objects.
+type TagService interface {
+	All() (Tag, error)
+
+	// Get retrieves the digest for the provided tag.
+	Get(name string) (Tag, error)
```

Comment:
Mostly referring to this conversation https://github.com/docker/distribution/pull/179/files#r24955886
Just trying to leave that as an option, it may be desirable that the external trust system completely disregards the tags interface on the registry in place of its own system.